src/librustc/ich/impls_mir.rs

```diff
@@ -29,7 +29,7 @@ impl_stable_hash_for!(struct mir::UnsafetyCheckResult { violations, unsafe_block
 
 impl_stable_hash_for!(enum mir::BorrowKind {
     Shared,
-    Shallow,
+    Guard,
```

Comment:
the review would probably be a little easier if you factored out a commit that held just the alpha-rename of `Shallow` to `Guard`